providers/sendinblue/src/lib/sendinblue.provider.ts

```diff
@@ -54,6 +56,75 @@ export class SendinblueEmailProvider implements IEmailProvider {
     };
   }
 
+  getMessageId(body: any | any[]): string[] {
+    if (Array.isArray(body)) {
+      return body.map((item) => item['message-id']);
+    }
+
+    return [body['message-id']];
+  }
+
+  parseEventBody(
+    body: any | any[],
+    identifier: string
+  ): IEmailEventBody | undefined {
+    if (Array.isArray(body)) {
+      body = body.find((item) => item['message-id'] === identifier);
+    }
+
+    if (!body) {
+      return undefined;
+    }
+
+    const status = this.getStatus(body.event);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: body.date,
+      externalId: body.id,
+      row: body,
+    };
+  }
+
+  private getStatus(event: string): EmailEventStatusEnum | undefined {
+    switch (event) {
+      case 'sent':
```

Comment:
Maybe these can be an internal enum inside of this file to make easy to change in the future?

Reply:
I believe that is contradictory to what we have [discussed recently](https://discord.com/channels/895029566685462578/937309213036785744/1029245883499429899).

However, I do agree and much prefer creating a new `SendinblueEventStatusEnum`.